cvm/vllm_proxy/proxy_app.py

```diff
@@ -0,0 +1,182 @@
+
+
+@app.post("/v1/chat/completions")
+async def chat(req: Request):
+    # expected : {"prompt": "...", "document": "...", "user_id": "..."}
+    dt_call2 = None
+    nb_query = 1
+    global_t0 = time.perf_counter()
+
+    try:
+        data = await req.json()
+    except Exception as e:
+        logger.error("❌ Invalid JSON: %s", e)
+        return JSONResponse({"error": "invalid JSON"}, status_code=400)
+
+    assert 'prompt' in data, '❌ Bad request keys'
+    assert 'document' in data, '❌ Bad request keys'
+    assert 'user_id' in data, '❌ Bad request keys'
+
+    prompt = data.get("prompt", "")
+    document = data.get("document", "")
+    user_id = data.get("user_id", "")
+    model = data.get("model", MODEL_ID)
+    temperature = data.get("temperature", DEFAULT_TEMPERATURE)
+    max_tokens = data.get("max_tokens", DEFAULT_MAX_TOKENS)
+    prompt_merged = f"{prompt}\n\nDocument:\n{document}" if document else prompt
+
+    payload1 = {
+        "user_id": user_id,
+        "model": model,
+        "messages": [
+            {"role": "system", "content": BASE_SYSTEM_CORE + '\n' + IS_TEE_RELATED_PROMPT},
+            {"role": "user",   "content": prompt_merged}
+        ],
+        "extra_body": {"cache_salt": f"user:{user_id}"},
+        "temperature": temperature,
+        "max_tokens": max_tokens,
+    }
+
+    t0 = time.perf_counter()
+    async with httpx.AsyncClient(timeout=httpx.Timeout(10.0, read=120.0)) as c:
+        final_r = await c.post(f"{URL_BASE_VLLM}/v1/chat/completions", json=payload1)
+        dt_call1 = time.perf_counter() - t0
+        r1 = final_r.json()
+        logger.info(f"⏱️ [[Endpoint | /v1/chat/completions]] Call1 in {dt_call1}s")
+        content1 = r1["choices"][0]["message"]["content"].strip()
+        reasoning_content1 = r1["choices"][0]["message"]["reasoning_content"].strip()
+        logger.info(f"1️⃣💬 vLLM response:\n {content1}")
+        logger.info(f"1️⃣🧠 vLLM reasoning_content1:\n {reasoning_content1}")
+
+        if content1 == "<confidential chat>":
+            nb_query += 1
+            payload2 = {
+                "user_id": user_id,
+                "model": model,
+                "messages": [
+                    {"role": "system", "content": BASE_SYSTEM_CORE + '\n' + TEE_INSTRUCTION},
+                    {"role": "user", "content": prompt_merged},
+                ],
+                "temperature": temperature,
+                "max_tokens": max_tokens,
+            }
+
+            t0 = time.perf_counter()
+            final_r = await c.post(f"{URL_BASE_VLLM}/v1/chat/completions", json=payload2)
+            dt_call2 = time.perf_counter() - t0
+
+            r2 = final_r.json()
+            content2 = r2["choices"][0]["message"]["content"].strip()
+            reasoning_content2 = r2["choices"][0]["message"]["reasoning_content"].strip()
+
+            logger.info(f"⏱️ [[Endpoint | /v1/chat/completions]] Call2 in {dt_call2}s")
+            logger.info(f"2️⃣💬 vLLM response:\n {content2}")
```

Comment:
I would start deleting all sensitive log. We shouldn't put anything sensitive in our logs.